Source/WebCore/dom/TrustedTypePolicy.cpp

```diff
@@ -50,65 +50,74 @@ TrustedTypePolicy::TrustedTypePolicy(const String& name, const TrustedTypePolicy
 
-        if (callbackResult.type() == CallbackResultType::Success) {
-            auto contents = callbackResult.releaseReturnValue();
+    if (policyValue.hasException())
+        return policyValue.releaseException();
 
-            return TrustedHTML::create(contents);
-        }
-        if (callbackResult.type() == CallbackResultType::ExceptionThrown)
-            return Exception { ExceptionCode::ExistingExceptionError };
-    }
-
-    return Exception {
-        ExceptionCode::TypeError,
-        makeString("Policy "_s, m_name,
-        "'s TrustedTypePolicyOptions did not specify a 'createHTML' member."_s)
-    };
+    return TrustedHTML::create(policyValue.releaseReturnValue());
 }
 
 ExceptionOr<Ref<TrustedScript>> TrustedTypePolicy::createScript(const String& input, FixedVector<JSC::Strong<JSC::Unknown>>&& arguments)
 {
-    if (m_createScriptCallback) {
-        auto callbackResult = m_createScriptCallback->handleEvent(input, WTFMove(arguments));
+    auto policyValue = getPolicyValue("TrustedScript"_s, input, WTFMove(arguments));
 
-        if (callbackResult.type() == CallbackResultType::Success) {
-            auto contents = callbackResult.releaseReturnValue();
+    if (policyValue.hasException())
+        return policyValue.releaseException();
 
-            return TrustedScript::create(contents);
-        }
-        if (callbackResult.type() == CallbackResultType::ExceptionThrown)
-            return Exception { ExceptionCode::ExistingExceptionError };
-    }
-
-    return Exception {
-        ExceptionCode::TypeError,
-        makeString("Policy "_s, m_name,
-        "'s TrustedTypePolicyOptions did not specify a 'createScript' member."_s)
-    };
+    return TrustedScript::create(policyValue.releaseReturnValue());
 }
 
 ExceptionOr<Ref<TrustedScriptURL>> TrustedTypePolicy::createScriptURL(const String& input, FixedVector<JSC::Strong<JSC::Unknown>>&& arguments)
 {
-    if (m_createScriptURLCallback) {
-        auto callbackResult = m_createScriptURLCallback->handleEvent(input, WTFMove(arguments));
+    auto policyValue = getPolicyValue("TrustedScriptURL"_s, input, WTFMove(arguments));
+
+    if (policyValue.hasException())
+        return policyValue.releaseException();
 
-        if (callbackResult.type() == CallbackResultType::Success) {
-            auto contents = callbackResult.releaseReturnValue();
+    return TrustedScriptURL::create(policyValue.releaseReturnValue());
+}
 
-            return TrustedScriptURL::create(contents);
-        }
-        if (callbackResult.type() == CallbackResultType::ExceptionThrown)
-            return Exception { ExceptionCode::ExistingExceptionError };
+// https://w3c.github.io/trusted-types/dist/spec/#get-trusted-type-policy-value-algorithm
+ExceptionOr<String> TrustedTypePolicy::getPolicyValue(const String& trustedTypeName, const String& input, FixedVector<JSC::Strong<JSC::Unknown>>&& arguments, bool throwIfMissing)
+{
+    auto callbackName = emptyString();
```

Comment:
Because the 3 callbacks are all of different types I don't know how to do this without using a variant which would make the code needlessly complicated imo?